inst/cpp/sim.cpp

```diff
@@ -76,7 +76,8 @@ List sim_wrapper_cpp (NumericVector A, List design, List par, NumericVector iov_
   // insert_parameter_definitions
 
   // Initialize parameters, compartments, etc:
-  pk_code(0, times, doses, 0, dose_cmt, dose_type, iov_bin);
+
+  pk_code(0, times, doses, doses[0], dose_cmt, dose_type, iov_bin);
```

Comment:
this fix, and the one a few lines below, are required to make sure `prv_dose` has the correct value at the first row of the event table for the simulation.